cmake/package_general.cmake

```diff
@@ -1,86 +1,140 @@
+set(CPACK_COMPONENT_LIBRARY_DESCRIPTION         "SmartPeak libraries for building applications against")
+set(CPACK_COMPONENT_APPLICATIONS_DESCRIPTION    "SmartPeak executables to run the GUI or scripts")
+set(CPACK_COMPONENT_DOC_DESCRIPTION             "Class documentation with tutorials")
+set(CPACK_COMPONENT_SHARE_GROUP                 "Development")
+set(CPACK_COMPONENT_LIBRARY_GROUP               "Runtime")
+set(CPACK_COMPONENT_APPLICATIONS_GROUP          "Runtime")
+set(CPACK_COMPONENT_DOC_GROUP                   "Runtime")
+set(CPACK_ALL_INSTALL_TYPES                     Full Developer)
+set(CPACK_COMPONENT_SHARE_INSTALL_TYPES         Developer Full)
+set(CPACK_COMPONENT_LIBRARY_INSTALL_TYPES       Full)
+set(CPACK_COMPONENT_APPLICATIONS_INSTALL_TYPES  Full)
+set(CPACK_COMPONENT_DOC_INSTALL_TYPES           Full)
+
+
+if (APPLE)
+
+  set(CPACK_GENERATOR                   "DragNDrop")
+  set(BUNDLE_OUTPUT_PATH                "${CMAKE_BINARY_DIR}/bin/GUI.app")
+  set(CPACK_MONOLITHIC_INSTALL          ON)
+  set(CPACK_BUNDLE_NAME                 "SmartPeak")
+  set(CPACK_BUNDLE_PLIST                ${PROJECT_SOURCE_DIR}/cmake/MacOSX/Info.plist)
+  set(CPACK_BUNDLE_ICON                 ${PROJECT_SOURCE_DIR}/cmake/MacOSX/sp.icns)
+
+elseif(UNIX AND CMAKE_SYSTEM_NAME MATCHES Linux)
+
+  set(CPACK_GENERATOR                   "DEB")
+  set(CPACK_DEBIAN_PACKAGE_MAINTAINER   "The SmartPeak Team")
+  set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS    ON)
+  set(CPACK_DEBIAN_PACKAGE_NAME         "SmartPeak")
+  set(CPACK_DEBIAN_PACKAGE_ARCHITECTURE "amd64")
+  set(CPACK_DEBIAN_ARCHIVE_TYPE         "gnutar")
+  set(CPACK_DEBIAN_PACKAGE_PRIORITY     "optional")
+  set(CPACK_DEBIAN_PACKAGE_SECTION      "science")
+  set(CPACK_MONOLITHIC_INSTALL          ON)
+
+endif()
+
+
 find_package(SDL2 REQUIRED)
 if (SDL2_FOUND)
-  # Add the directory for later calls to CPACK
-  string(FIND ${SDL2_LIBRARY} ${VC_LIB_PATH_SUFFIX} SDL2_library_suffix_pos)
-  string(SUBSTRING ${SDL2_LIBRARY} 0 ${SDL2_library_suffix_pos} SDL2_LIBRARY_DIR)
-  set(SDL2_LIBRARIES_DIR "${SDL2_LIBRARY_DIR}${VC_LIB_PATH_SUFFIX}")
+  if (WIN32)
+    # Add the directory for later calls to CPACK
+    string(FIND ${SDL2_LIBRARY} ${VC_LIB_PATH_SUFFIX} SDL2_library_suffix_pos)
+    string(SUBSTRING ${SDL2_LIBRARY} 0 ${SDL2_library_suffix_pos} SDL2_LIBRARY_DIR)
+    set(SDL2_LIBRARIES_DIR "${SDL2_LIBRARY_DIR}${VC_LIB_PATH_SUFFIX}")
+  elseif(APPLE)
+    set(SDL2_LIBRARIES_DIR "")
+	  string(REGEX REPLACE ";.*$" "" SDL2_LIB_DIR "${SDL2_LIBRARY}")
+	  if(EXISTS "${SDL2_LIB_DIR}/Versions/A/Frameworks")
+      set(SDL2_LIBRARIES_DIR  "${SDL2_LIB_DIR}/Versions/A")
+      set(HIDAPI_DIR          "${SDL2_LIB_DIR}/Versions/A/Frameworks")
+	  endif()
+  endif()
 endif (SDL2_FOUND)
+
 find_package(OpenMS REQUIRED)
 if (OpenMS_FOUND)
   ## Add the directory for later calls to CPACK
-  set(OpenMS_LIBRARY_DIR ${OpenMS_DIR}/bin/Release)
+  set(OpenMS_LIBRARY_DIR ${OpenMS_DIR}/lib)
```

Comment:
Can you confirm that this does not break the windows CPack?

Reply:
The `ddl`s inside the bin folder are needed for windows CPack so it has to remain unchanged.